src/main/cli/settings.c

```diff
@@ -1017,7 +1017,7 @@ const clivalue_t valueTable[] = {
 
     { PARAM_NAME_GPS_RESCUE_RETURN_ALT,      VAR_UINT8  | MASTER_VALUE, .config.minmaxUnsigned = { 2, 255 }, PG_GPS_RESCUE, offsetof(gpsRescueConfig_t, initialAltitudeM) },
     { PARAM_NAME_GPS_RESCUE_RETURN_SPEED,    VAR_UINT16 | MASTER_VALUE, .config.minmaxUnsigned = { 0, 3000 }, PG_GPS_RESCUE, offsetof(gpsRescueConfig_t, rescueGroundspeed) },
-    { PARAM_NAME_GPS_RESCUE_MAX_RESCUE_ANGLE, VAR_UINT8  | MASTER_VALUE, .config.minmaxUnsigned = { 0, 80 }, PG_GPS_RESCUE, offsetof(gpsRescueConfig_t, maxRescueAngle) },
+    { PARAM_NAME_GPS_RESCUE_MAX_RESCUE_ANGLE, VAR_UINT8  | MASTER_VALUE, .config.minmaxUnsigned = { 30, 60 }, PG_GPS_RESCUE, offsetof(gpsRescueConfig_t, maxRescueAngle) },
```

Comment:
What about this change? This can give people problems when flashing from 4.4.1 to 4.4.2. You can get settings outside of the allowed range.

Reply:
We should mention on release users need to use full chip erase as 4.4.1 will be pulled.

Reply:
But that's the whole idea... not having to use full chip erase on minor version updates...

Reply:
Okay I warned you guys several times now and always got ignored, so don't expect me to do support work for this mess. I'll approve and leave you with whatever result comes out of it.

Reply:
No you are right - we should be more strict on this issue.

Any suggestion how we would fix it as these changes are likely needed to tune the GPS rescue parameters.

Reply:
I am unhappy also about back-porting fixes.  As I've said, the optimal IMU adaptation solution is not yet 100% certain.  What we do know is that IMU gain manipulation can reduce the risk of failure when the IMU is disoriented.  We don't exactly know how best to implement this.  

In essence I am still developing the IMU adaptation code within 4.5, and it is not yet final.  

However, this IMU adaptation code does improve rescue reliability for 4.4 users, and that may save some people from losing their quad.  

If a person had set a non-standard pitch angle for GPS rescue, the outcome won't be any worse than it used to be.  If they save a preset they will get a warning when they apply their old settings.  

The default range is definitely best if it is constrained, at this point, to the 30-60 degree range; I cannot say if that range could be made wider, or if it should be narrower, until we know more about how the IMU works.

To that end I will make a basic PR for test purposes only, where the pilot can set the IMU gain to a fixed value that applies only in the return home phase of a rescue, so that we can determine better what IMU gain settings work best when the IMU is in error.  From that data we may be able to improve the automatic adjustment of the IMU gain, and perhaps then tolerate a wider range of max angle limit values.  

The question then becomes, should we keep back porting improvements to 4.4, or should they be for 4.5 only.  Generally I find back-porting to be annoying, it generates Configurator mismatches where values can be entered in Configurator that are outside the range preferred in the CLI, for example. 

I hope to have matters more clear in coming weeks, with the time frame depending on how much time I have, how many people can test IMU related code, and whether anyone else can provide tested code that improves matters.

Reply:
Simply reverting the range would solve the issue ???

Reply:
```suggestion
    { PARAM_NAME_GPS_RESCUE_MAX_RESCUE_ANGLE, VAR_UINT8  | MASTER_VALUE, .config.minmaxUnsigned = { 0, 80 }, PG_GPS_RESCUE, offsetof(gpsRescueConfig_t, maxRescueAngle) },
```

Reply:
> However, this IMU adaptation code does improve rescue reliability for 4.4 users, and that may save some people from losing their quad.

Yes that was the reason for the backport. We could leave the range however to not break configuration. Further improvements should go 4.5 only unless critical.

Reply:
> What about this change? This can give people problems when flashing from 4.4.1 to 4.4.2. You can get settings outside of the allowed range. Same problem [here](https://github.com/betaflight/betaflight/pull/12835/files#diff-1fca665faca91d927d814882e17ad63e404a95a51adc50b4b9aab0e74e160f72R1013) but the PR got merged anyway without addressing it for some reason... we cannot just change all the ranges as we like for old releases. It's starting to become a little mine field for corrupted settings 😬

One thing we can do here is if applying a setting outside of the range it should error - and alert the user. 

We could if the setting is below the range, raise a warning, and set it to the lower extreme - and vice versa?

Reply:
@blckmn I guess the problem is that we don't have any check in place to do that. After upgrading from 4.4.1 to 4.4.2 there will be no new settings applied. The old settings just stay in memory, so if you type `get xyz` in the CLI, the corrupted value will get displayed in the CLI without an error.

Maybe if we scan all settings and check the ranges after an update that would work but I don't know if this is worth the hassle when we just need to stay away from shrinking the range on maintenance PRs. Expanding the range works but shrinking it can be problematic.

Reply:
Fixed in https://github.com/betaflight/betaflight/pull/12854

Reply:
> Simply reverting the range would solve the issue ???

Yes, that's what I am trying to say 😀

Reply:
OK so i'll revert the range limit modification here.
No problem.

Reply:
Fixed already @ctzsnooze 

https://github.com/betaflight/betaflight/pull/12845#discussion_r1212407492

Reply:
IMO 30deg min limit is bad idea overall. It is designed to handle RTH without magnetometer. But larger drone with compass may use small roll safely.